crates/cosmos/cosmos-chain-components/src/impls/queries/counterparty_connection_id.rs

```diff
@@ -0,0 +1,53 @@
+use cgp::prelude::CanRaiseAsyncError;
+use hermes_relayer_components::chain::traits::queries::chain_status::CanQueryChainHeight;
+use hermes_relayer_components::chain::traits::queries::connection_end::CanQueryConnectionEnd;
+use hermes_relayer_components::chain::traits::queries::counterparty_connection_id::CounterpartyConnectionIdQuerier;
+use hermes_relayer_components::chain::traits::types::channel::HasChannelEndType;
+use hermes_relayer_components::chain::traits::types::ibc::HasConnectionIdType;
+use ibc::core::channel::types::channel::ChannelEnd;
+use ibc::core::connection::types::ConnectionEnd;
+use ibc::core::host::types::identifiers::ConnectionId;
+
+pub struct QueryCounterpartyConnectionId;
+
+impl<Chain, Counterparty> CounterpartyConnectionIdQuerier<Chain, Counterparty>
+    for QueryCounterpartyConnectionId
+where
+    Chain: CanQueryChainHeight
+        + HasConnectionIdType<Counterparty>
+        + HasChannelEndType<Counterparty, ChannelEnd = ChannelEnd>
+        + CanQueryConnectionEnd<
+            Counterparty,
+            ConnectionEnd = ConnectionEnd,
+            ConnectionId = ConnectionId,
+        > + CanRaiseAsyncError<String>,
+    Counterparty: HasConnectionIdType<Chain, ConnectionId = ConnectionId>,
+{
+    async fn query_channel_end_counterparty_connection_id(
+        chain: &Chain,
+        channel_end: &Chain::ChannelEnd,
+    ) -> Result<Counterparty::ConnectionId, Chain::Error> {
+        let connection_id = channel_end
+            .connection_hops
+            .first()
+            .ok_or_else(|| Chain::raise_error(format!("channel end has no connection_hops")))?;
```

Comment:
shouldn't we validate the `len()` is `1`? as we only support single connection hop.

Reply:
Fixed in 0d9ad5821e91e41c9fd36db4355830aeaa0e500f.